plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java

```diff
@@ -155,19 +156,15 @@ public void testInsertWriteBulkinessWithTimestamps(String timestampType)
         }
     }
 
-    @Test(dataProvider = "testTableNameDataProvider")
-    public void testCreateTableWithSpecialCharacterName(String tableName)
+    //TODO move test to BaseConnectorTest and merge this data provider with testColumnNameDataProvider
+    @Test(dataProvider = "testTableAndColumNameDataProvider")
+    public void testCreateAndDropTableWithSpecialCharacterName(String tableName)
     {
-        if (!hasBehavior(SUPPORTS_CREATE_TABLE)) {
-            assertQueryFails("CREATE TABLE " + "\"" + tableName + "\"" + " (a bigint, b double, c varchar(50))", "This connector does not support creating tables");
-            return;
-        }
-
-        assertUpdate("CREATE TABLE " + "\"" + tableName + "\"" + " (a bigint, b double, c varchar(50))");
-        assertTrue(getQueryRunner().tableExists(getSession(), tableName.replace("\"\"", "\"")));
-        assertTableColumnNames("\"" + tableName + "\"", "a", "b", "c");
+        assertUpdate("CREATE TABLE " + toTableNameInSql(tableName) + " (a bigint, b double, c varchar(50))");
```

Comment:
Capture `toTableNameInSql(tableName)` as a variable 
and let's inline that function.